### PR TITLE
[Tests] Add tests on WireRestShape component

### LINKED ISSUE #23: [Tests] Add Unit tests on main components

### PR DESCRIPTION
**Pull request by epernod**
Note: This branch is based on an old version of master from 10 Aug. To make sure no regression has been introduced since that date.

fix #23

### CODE REVIEW

File: BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    ///// Test creation of WireRestShape in a default scene and check parameters cohesion 
+    void testParameterInit();
+
+    ///// Test creation of WireRestShape in a default scene and check created topology 
+    //void testTopologyInit();
+
+    //// TODO test on topology init from MeshLoader
+
+    ///// Test WireRestShape transform methods 
+    //void testTransformMethods();
```
Comment (alxbilger) [inline BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp:94]: To remove

File: BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        "   </Node>                                                                             "
+        "</Node>                                                      ";
+
+    EXPECT_MSG_NOEMIT(Error);
+    loadScene(scene);
+
+    InterventionalRadioCtrlRig3::SPtr IRCtrlPtr = m_root->get< InterventionalRadioCtrlRig3 >(sofa::core::objectmodel::BaseContext::SearchDown);
+    EXPECT_NE(IRCtrlPtr.get(), nullptr);
+    //TODO: change component to match behavior expected by tests
+    //EXPECT_EQ(IRCtrlPtr->getComponentState(), ComponentState::Valid);
```
Comment (alxbilger) [inline BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp:150]: How complex it is? Can it be done quickly in this PR? This is to not forget later

File: BeamAdapter_test/component/model/WireRestShape_test.cpp [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    /// Test creation of WireRestShape in a default scene with needed Topology components
+    void testDefaultInit();
+
+    /// Test creation of WireRestShape in a default scene and check parameters cohesion 
+    void testParameterInit();
+
+    /// Test creation of WireRestShape in a default scene and check created topology 
+    void testTopologyInit();
+
+    // TODO test on topology init from MeshLoader
```
Comment (alxbilger) [inline BeamAdapter_test/component/model/WireRestShape_test.cpp:89]: To remove
```suggestion
```

**Review by alxbilger — CHANGES_REQUESTED** [review]

File: BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp (hunk 2 above)
Comment (epernod) [reply BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp:150]: It is not complex at all. In fact it has already been done in a pending PR.

The idea was to create the tests based on an old version of the plugin before all refactoring PR. Thus we make sure we didn't change the behavior with the recent PR.
And changing the componentState is part of one of the refactoring PR. I will rebase and activate the test in it.

File: BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp (hunk 1 above)
Comment (epernod) [inline BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp:94]:
```suggestion
    /** TODO epernod 2023-05-03: 
     - Test creation of WireRestShape in a default scene and check created topology 
     - Test WireRestShape topology init from MeshLoader
     - Test WireRestShape transform methods 
     */   
```

File: BeamAdapter_test/CMakeLists.txt [hunk 4]
```diff
@@ -8,8 +8,10 @@ find_package(Threads REQUIRED)
 
 set(SOURCE_FILES
     BeamInterpolation_test.cpp
+    component/controller/InterventionalRadiologyController_test.cpp
     component/constraint/AdaptiveBeamSlidingConstraintTest.cpp
     component/forcefield/AdaptiveBeamForceFieldAndMassTest.cpp
+	component/model/WireRestShape_test.cpp
```
Comment (epernod) [inline BeamAdapter_test/CMakeLists.txt:14]:
```suggestion
    component/forcefield/AdaptiveBeamForceFieldAndMassTest.cpp
    component/model/WireRestShape_test.cpp
```

File: BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp (hunk 2 above)
Comment (epernod) [reply BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp]: forget my previous answer. The component has been updated and it is now tested

File: BeamAdapter_test/component/model/WireRestShape_test.cpp [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+        "   </Node>                                                   "
+        "</Node>                                                      ";
+
+    loadScene(scene);
+
+
+    WireRestShapeRig3::SPtr wireRShape = this->m_root->get< WireRestShapeRig3 >(sofa::core::objectmodel::BaseContext::SearchDown);
+    EXPECT_NE(wireRShape.get(), nullptr);
+    //TODO: change component to match behavior expected by tests
+    //EXPECT_EQ(wireRShape->getComponentState(), ComponentState::Valid);
```
Comment (alxbilger) [inline BeamAdapter_test/component/model/WireRestShape_test.cpp:140]: What about this code?

File: BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    void testDefaultInit();
+
+    ///// Test creation of WireRestShape in a default scene and check parameters cohesion 
+    void testParameterInit();
+
+    /** TODO epernod 2023-05-03: 
+     - Test creation of WireRestShape in a default scene and check created topology 
+     - Test WireRestShape topology init from MeshLoader
+     - Test WireRestShape transform methods 
+     */
```
Comment (alxbilger) [inline BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp:92]: Do you want to keep that?

Comment (epernod) [reply BeamAdapter_test/component/controller/InterventionalRadiologyController_test.cpp:92]: yes it is a todo :)

**Review by alxbilger — APPROVED** [review]

**Pull request merged**